### ISSUE TITLE
Keyboard drops on the device

### ISSUE THREAD

**Issue by dougbarr** [kawanet/KWFontPicker#2]
FANTASTIC CODE!  Overall it works great!  I am experiencing one issue tho. In simulator, everything works perfectly, however, when I actually try it on my device, while I'm typing, the keyboard frequently just drops. I haven't tested it on another device (I'm on iOS6.1, retina iPad).  It appears to be when I move the device (like I tilt it for better positioning to type or something). This causes a bad user experience.  Is there a method to stop the keyboard from dropping down like it does?

**Comment by kawanet** [comment]
The keyboard will be closed when it detects an device rotation. Is this the case of you?
By the way, the code is rarely tested on iPad device as it's originally written for iPhone.
I hope I could work on iPad as well, however.

**Comment by dougbarr** [comment]
No, it doesn't even have to be a rotation.  Let's say the device is in landscape, if you just tilt the device, it closes the keyboard.  I think the issue may be in either: 
- (BOOL)textViewShouldEndEditing:(UITextView *)textView  OR
- (BOOL)textViewDidEndEditing:(UITextView *)textView

**Comment by dougbarr** [comment]
textViewShouldEndEditing is definitely being called when the device tilts or rotates. Do you know how to prevent?

**Comment by dougbarr** [comment]
I solved it.  You have an observer when the device orientation changes and you close the text editor:
- (void) deviceOrientationDidChange:(NSNotification *)notification
  {
  // [self closeTextEditor];
  }
